xalan/src/main/java/org/apache/xalan/Version.java

```diff
@@ -51,12 +49,24 @@ public class Version
   private static boolean snapshot;
 
   static {
-    readProperties();
-    parseVersionNumber();
+    parseVersionNumber(readVersionNumber());
+  }
+
+  static String readVersionNumber() {
+    try (InputStream fromResource = getPropertiesStream()) {
+      Properties properties = new Properties();
+      if (fromResource != null) {
+        properties.load(fromResource);
+        return properties.getProperty("version", NO_VERSION);
+      }
+    } catch (Exception e) {
+      new RuntimeException("Cannot read properties file to extract Xalan version number information: ", e)
+        .printStackTrace();
+    }
```

Comment:
It might be a slightly better idea to use `catch(Throwable)` since `Exception` does not cover all the exception types.

Reply:
`Exception` does cover all exception types, just not all throwable types. For instance, `Error`s ought not be caught. They are supposed to be escalated and ought to exit the application, which is why I am only catching exceptions.

Reply:
Technically, Java calls them "exceptions", not "throwables": https://docs.oracle.com/javase/specs/jls/se21/html/jls-11.html

> **Every exception** is represented by an instance of the class **`Throwable`** or one of its subclasses ([§11.1](https://docs.oracle.com/javase/specs/jls/se21/html/jls-11.html#jls-11.1))

Reply:
Stop nitpicking obsessively, won't you? You explicitly asked me to catch `Throwable`, I explained why I won't, because that would encompass fatal `Error` types.